src/Functions/EntropyLeardnedHash.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+// Reference to the paper about entropy-learned hashing:
+// https://bhentsch.github.io/doc/EntropyLearnedHashing.pdf
+
+#include <cstdint>
+#include <limits>
+#include <map>
+#include <string>
+#include <vector>
+
+// Key is a byte string
+namespace EntropyLearnedHashing
+{
+using Key = std::string;
+
+[[nodiscard]] Key getPartialKey(const std::string & key, const std::vector<size_t> & positions);
+
+class IDsManager
+{
+private:
+    static IDsManager * instance_ptr;
+    IDsManager() { }
+    IDsManager(const IDsManager &);
+    IDsManager & operator=(IDsManager &);
+
+public:
+    static IDsManager * getInstance()
```

Comment:
For a singleton, you could just do

``` cpp
static IDsManager & instance() 
{                                       
    static IDsManager instance;
    return instance;                    
}                                       
```

(local static objects are guaranteed to be thread-safe since C++11)

Let's also move this into the source file.